### ISSUE TITLE
'AsyncQueueWorker': identifier not found on Windows 10 x64

### ISSUE THREAD

**Issue by jviotti** [addaleax/lzma-native#19]
Hi @addaleax ,

Since [this commit](https://github.com/addaleax/lzma-native/commit/86def069a0316866b03b46f4e92c072bfbfac3f9), which introduces `AsyncQueueWorked`, I get the following error when building Windows:

```
..\src\lzma-stream.cpp(182): error C3861: 'AsyncQueueWorker': identifier not found [C:\Projects\etcher\node_modules\lzm
a-native\build\lzma_native.vcxproj]
```

As far as I can see, this comes from [NAN](https://github.com/nodejs/nan). Any ideas?

**Comment by addaleax** [comment]
@jviotti What Node.js version are you using? Would this fix it for you?

```diff
diff --git a/src/lzma-stream.cpp b/src/lzma-stream.cpp
index b2eb8ea1416a..12a17656c796 100644
--- a/src/lzma-stream.cpp
+++ b/src/lzma-stream.cpp
@@ -179,7 +179,7 @@ NAN_METHOD(LZMAStream::Code) {
   
   if (async) {
 #ifdef LZMA_ASYNC_AVAILABLE
-    AsyncQueueWorker(new LZMAStreamCodingWorker(self));
+    Nan::AsyncQueueWorker(new LZMAStreamCodingWorker(self));
 #else
     std::abort();
 #endif
```

**Comment by addaleax** [comment]
It’s weird that I can’t reproduce this locally on Linux or Windows and CI didn’t pick that up either, but I’m guessing that must be it.

@jviotti Can you verify that the above change works? Sorry for all the compilation trouble here…

**Comment by jviotti** [comment]
Hi @addaleax ,

Thanks for the quick response! I'm running v5.10.0, let me apply the patch and see how it goes.

**Comment by jviotti** [comment]
The patch works fine, but now I get a different error:

```
C:\Projects\lzma-native>npm install --build-from-source
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.

> lzma-native@1.4.1 install C:\Projects\lzma-native
> node-pre-gyp install --fallback-to-build


C:\Projects\lzma-native>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" clean )  else (node "" clean )

C:\Projects\lzma-native>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" configure --fallback-to-build --module=C:\Projects\lzma-native\binding\lzma_native.node --module_name=lzma_native --module_path=C:\Projects\lzma-native\binding )  else (node "" configure --fallback-to-build --module=C:\Projects\lzma-native\binding\lzma_native.node --module_name=lzma_native --module_path=C:\Projects\lzma-native\binding )
Warning: Missing input files:
C:\Projects\lzma-native\build\..\deps\doc\liblzma.def

C:\Projects\lzma-native>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" build --fallback-to-build --module=C:\Projects\lzma-native\binding\lzma_native.node --module_name=lzma_native --module_path=C:\Projects\lzma-native\binding )  else (node "" build --fallback-to-build --module=C:\Projects\lzma-native\binding\lzma_native.node --module_name=lzma_native --module_path=C:\Projects\lzma-native\binding )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  build
  Microsoft (R) Library Manager Version 12.00.31101.0
  Copyright (C) Microsoft Corporation.  All rights reserved.

LINK : fatal error LNK1104: cannot open file 'C:\Projects\lzma-native\deps\doc\liblzma.def' [C:\Projects\lzma-native\build\liblzma.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `msbuild` failed with exit code: 1
```

The mentioned file indeed doesn't exist.

**Comment by addaleax** [comment]
You may have to run `npm prepare` first for compiling on Windows :/

**Comment by jviotti** [comment]
That was it, works like a charm now! Thanks a lot again for the awesome support!

**Comment by addaleax** [comment]
Sure. I’ve committed the fix in e16749c644aec3bafbb442f50f892ddbdc135054 and I’ll publish 1.4.1 once the CI runs have completed… :snail:

**Comment by addaleax** [comment]
Published as 1.4.1 … thanks for the report!

**Comment by jviotti** [comment]
All up and running with the new version, thanks again!
